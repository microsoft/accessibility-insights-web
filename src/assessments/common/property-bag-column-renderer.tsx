// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { InstanceTableRow } from 'assessments/types/instance-table-data';
import { ColumnValueBag } from 'common/types/property-bag/column-value-bag';
import { PropertyBagColumnRendererConfig } from 'common/types/property-bag/property-bag-column-renderer-config';
import { isEmpty } from 'lodash';
import * as React from 'react';

import { DictionaryStringTo } from 'types/common-types';

export function propertyBagColumnRenderer<TPropertyBag extends ColumnValueBag>(
    item: InstanceTableRow<TPropertyBag>,
    configs: PropertyBagColumnRendererConfig<TPropertyBag>[],
): JSX.Element {
    const mapper = (config: PropertyBagColumnRendererConfig<TPropertyBag>, index: number) => {
        if (item.instance.propertyBag == null) {
            return null;
        }
        const value = item.instance.propertyBag[config.propertyName];
        if (value == null && config.defaultValue == null) {
            return null;
        }

        return render(config, value, index);
    };

    const render = (
        config: PropertyBagColumnRendererConfig<TPropertyBag>,
        value: any,
        index: number,
    ) => {
        return (
            <div key={`property-${index}`} className="property-bag-div">
                <span className="display-name">{`${config.displayName}: `}</span>
                {renderValue(config, value)}
            </div>
        );
    };

    const renderValue = (config: PropertyBagColumnRendererConfig<TPropertyBag>, value: any) => {
        if (config.expand) {
            return renderProperties(config, value);
        }

        return <React.Fragment>{value || config.defaultValue}</React.Fragment>;
    };

    const renderProperties = (
        config: PropertyBagColumnRendererConfig<TPropertyBag>,
        propertyMap: DictionaryStringTo<string>,
    ) => {
        if (isEmpty(propertyMap)) {
            return <React.Fragment>{config.defaultValue.toString()}</React.Fragment>;
        }

        return Object.keys(propertyMap).map(key => {
            return (
                <div key={key} className="expanded-property-div">
                    {renderInnerKeyValue(key, propertyMap[key])}
                </div>
            );
        });
    };

    const renderInnerKeyValue = (key: string, value: string) => {
        if (value === null) {
            return <span className="display-name">{`${key}`}</span>;
        } else {
            return (
                <React.Fragment>
                    <span className="display-name">{`${key}: `}</span>
                    {value}
                </React.Fragment>
            );
        }
    };

    return <div className="property-bag-container">{configs.map(mapper)}</div>;
}
