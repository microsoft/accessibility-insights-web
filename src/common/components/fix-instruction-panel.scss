// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import '../styles/colors.scss';

.insights-fix-instruction-list {
    li {
        span.fix-instruction-color-box {
            width: 14px;
            height: 14px;
            display: inline-block;
            vertical-align: -5%;
            margin: 0 2px;
            border: 1px solid $always-black;
        }
    }

    .screen-reader-only {
        position: absolute;
        left: -10000px;
        top: auto;
        height: 1px;
        overflow: hidden;
        clip-path: inset(100% 100% 100% 100%);
    }
}
