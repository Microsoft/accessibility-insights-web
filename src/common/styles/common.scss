// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import 'high-contrast-override.scss';
@import 'colors.scss';
@import 'fonts.scss';

$focus-ring-color: $neutral-100;

@mixin set-focus-ring-styles($property: 'border') {
    #{$property}: 1px solid $focus-ring-color;
    @include ms-high-contrast-override {
        #{$property}: 1px dashed windowtext !important;
    }
}

.ms-Fabric.is-focusVisible .ms-Nav-group .ms-nav-linkButton:focus::after {
    @include set-focus-ring-styles('border');
}

button::after {
    content: '';
    position: absolute;
}

.ms-Button.ms-Button--primary:not(.is-disabled) {
    background: $communication-primary;

    &:hover {
        background: $communication-shade-10;
    }
}

.ms-fontColor-neutralPrimary,
.ms-fontColor-neutralPrimary--hover:hover {
    color: $neutral-100 !important;
}

.ms-Fabric {
    color: $neutral-100;
}

.generic-toggle-component {
    margin-bottom: 4vh;
    .toggle-container {
        margin-bottom: 1vh;
        .toggle-name {
            display: inline-block;
            font-size: 17px;
            font-weight: bold;
        }

        .toggle {
            float: right;
        }
    }
}

.high-contrast-theme {
    .insights-link {
        &:hover {
            text-decoration: underline;
        }
    }
    .is-selected {
        .status-icon {
            color: $high-contrast-icon-color !important;
        }
    }
}

.insights-link {
    // marking important to override office fabric style
    color: $link !important;
    text-decoration: none;

    &:hover {
        color: $link-hover !important;
        text-decoration: underline;
    }
}

.ms-Spinner {
    .ms-Spinner-circle {
        border-top-color: $spinner-text;
    }

    .ms-Spinner-label {
        color: $spinner-text;
    }
}

$normalTitleFontWeight: 300 !default;

$pivotItemLeftBorderWidth: 3px !default;
$pivotItemBorderStyle: solid !default;

$detailsViewHeaderBarHeight: 40px !default;
$detailsViewNavPivotsHeight: 41px !default;
$detailsViewCommandBarHeight: 40px !default;
$detailsViewCommandBarBorderHeight: 1px !default;
$detailsViewLeftNavWidth: 200px !default;
$detailsViewTotalHeaderHeight: (#{$detailsViewCommandBarBorderHeight} + #{$detailsViewCommandBarHeight} + #{$detailsViewHeaderBarHeight});

$fastPassRightPanelMarginRight: 24px !default;
$fastPassRightPanelMarginLeft: 24px !default;
$fastPassRightPanelMarginTop: 24px !default;

@mixin ellipsedText {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}
