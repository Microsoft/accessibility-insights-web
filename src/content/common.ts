// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as react from 'react';

import { link } from './link';
import { ContentCreator } from '../views/content/content-page';
import { GuidanceTitle as gt } from '../views/content/guidance-title';

export const create = ContentCreator(link);
export const React = react;
export const GuidanceTitle = gt;
