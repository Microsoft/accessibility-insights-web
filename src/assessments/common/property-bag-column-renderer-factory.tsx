// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { ColumnValueBag } from '../../common/types/property-bag/column-value-bag';
import { IAssessmentInstanceRowData } from '../../DetailsView/components/assessment-instance-table';
import { propertyBagColumnRenderer, PropertyBagColumnRendererConfig } from './property-bag-column-renderer';

export class PropertyBagColumnRendererFactory {
    public static get<TPropertyBag extends ColumnValueBag>(
        configs: PropertyBagColumnRendererConfig<TPropertyBag>[],
    ): (item: IAssessmentInstanceRowData<TPropertyBag>) => JSX.Element {
        return item => {
            return propertyBagColumnRenderer(item, configs);
        };
    }
}
