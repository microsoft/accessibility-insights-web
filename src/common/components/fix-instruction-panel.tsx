// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { FixInstructionProcessor } from 'common/components/fix-instruction-processor';
import { RecommendColor } from 'common/components/recommend-color';
import { CheckType } from 'common/types/check-type';
import * as React from 'react';
import { NamedFC } from '../react/named-fc';
import styles from './fix-instruction-panel.scss';

export interface FixInstructionPanelDeps {
    fixInstructionProcessor: FixInstructionProcessor;
    recommendColor: RecommendColor;
}

export interface FixInstructionPanelProps {
    deps: FixInstructionPanelDeps;
    checkType: CheckType;
    checks: { message: string }[];
    renderTitleElement: (titleText: string) => JSX.Element;
}

export const FixInstructionPanel = NamedFC<FixInstructionPanelProps>(
    'FixInstructionPanel',
    props => {
        const { fixInstructionProcessor } = props.deps;

        const getPanelTitle = (checkType: CheckType, checkCount: number): string => {
            if (checkCount === 1) {
                return 'Fix the following:';
            }
            if (checkType === CheckType.Any) {
                return 'Fix ONE of the following:';
            } else {
                return 'Fix ALL of the following:';
            }
        };

        const renderInstructions = (checkType: CheckType): JSX.Element[] => {
            const instructionList = props.checks.map((check, checkIndex) => {
                return (
                    <li key={`instruction-${CheckType[checkType]}-${checkIndex + 1}`}>
                        {fixInstructionProcessor.process(check.message, props.deps.recommendColor)}
                    </li>
                );
            });

            return instructionList;
        };

        if (props.checks.length === 0) {
            return null;
        }

        const title: string = getPanelTitle(props.checkType, props.checks.length);

        return (
            <div>
                {props.renderTitleElement(title)}
                <ul className={styles.insightsFixInstructionList}>
                    {renderInstructions(props.checkType)}
                </ul>
            </div>
        );
    },
);
