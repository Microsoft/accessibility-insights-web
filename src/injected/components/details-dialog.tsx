// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { DefaultButton, PrimaryButton } from 'office-ui-fabric-react/lib/Button';
import { Dialog, DialogType } from 'office-ui-fabric-react/lib/Dialog';
import { Icon } from 'office-ui-fabric-react/lib/Icon';
import * as React from 'react';

import { ClientBrowserAdapter } from '../../common/client-browser-adapter';
import { CopyIssueDetailsButton, CopyIssueDetailsButtonDeps } from '../../common/components/copy-issue-details-button';
import { FileIssueDetailsButton, FileIssueDetailsButtonDeps } from '../../common/components/file-issue-details-button';
import { FlaggedComponent } from '../../common/components/flagged-component';
import { NewTabLink } from '../../common/components/new-tab-link';
import { FeatureFlags } from '../../common/feature-flags';
import { IBaseStore } from '../../common/istore';
import { DevToolActionMessageCreator } from '../../common/message-creators/dev-tool-action-message-creator';
import { CreateIssueDetailsTextData } from '../../common/types/create-issue-details-text-data';
import { DevToolState } from '../../common/types/store-data/idev-tool-state';
import { UserConfigurationStoreData } from '../../common/types/store-data/user-configuration-store';
import { DecoratedAxeNodeResult } from '../scanner-utils';
import { TargetPageActionMessageCreator } from '../target-page-action-message-creator';
import { DetailsDialogHandler } from './../details-dialog-handler';
import { FixInstructionPanel } from './fix-instruction-panel';

export enum CheckType {
    All,
    Any,
    None,
}

export type DetailsDialogDeps = CopyIssueDetailsButtonDeps &
    FileIssueDetailsButtonDeps & {
        targetPageActionMessageCreator: TargetPageActionMessageCreator;
        clientBrowserAdapter: ClientBrowserAdapter;
    };

export interface DetailsDialogProps {
    deps: DetailsDialogDeps;
    userConfigStore: IBaseStore<UserConfigurationStoreData>;
    elementSelector: string;
    failedRules: IDictionaryStringTo<DecoratedAxeNodeResult>;
    target: string[];
    dialogHandler: DetailsDialogHandler;
    devToolStore: IBaseStore<DevToolState>;
    devToolActionMessageCreator: DevToolActionMessageCreator;
    featureFlagStoreData: IDictionaryStringTo<boolean>;
    devToolsShortcut: string;
}

export interface DetailsDialogState {
    showDialog: boolean;
    currentRuleIndex: number;
    canInspect: boolean;
    issueTrackerPath: string;
}

export class DetailsDialog extends React.Component<DetailsDialogProps, DetailsDialogState> {
    private onHideDialog: () => void;
    public onClickInspectButton: (ev) => void;
    private onLayoutDidMount: () => void;
    public onClickNextButton: () => void;
    public onClickBackButton: () => void;
    public isBackButtonDisabled: () => boolean;
    public isNextButtonDisabled: () => boolean;
    public isInspectButtonDisabled: () => boolean;

    constructor(props: DetailsDialogProps) {
        super(props);

        this.onHideDialog = () => {
            this.props.dialogHandler.hideDialog(this);
        };
        this.onClickNextButton = () => {
            this.props.dialogHandler.nextButtonClickHandler(this);
        };
        this.onClickBackButton = () => {
            this.props.dialogHandler.backButtonClickHandler(this);
        };
        this.onClickInspectButton = (ev: React.SyntheticEvent<MouseEvent>) => {
            this.props.dialogHandler.inspectButtonClickHandler(this, ev);
        };
        this.onLayoutDidMount = () => {
            this.props.dialogHandler.onLayoutDidMount();
        };
        this.componentDidMount = () => {
            this.props.dialogHandler.componentDidMount(this);
        };
        this.componentWillUnmount = () => {
            this.props.dialogHandler.componentWillUnmount(this);
        };
        this.onClickNextButton = () => {
            this.props.dialogHandler.nextButtonClickHandler(this);
        };
        this.onClickBackButton = () => {
            this.props.dialogHandler.backButtonClickHandler(this);
        };
        this.isBackButtonDisabled = () => {
            return this.props.dialogHandler.isBackButtonDisabled(this);
        };
        this.isNextButtonDisabled = () => {
            return this.props.dialogHandler.isNextButtonDisabled(this);
        };
        this.isInspectButtonDisabled = () => {
            return this.props.dialogHandler.isInspectButtonDisabled(this);
        };

        this.state = {
            showDialog: true,
            currentRuleIndex: 0,
            canInspect: true,
            issueTrackerPath: '',
        };
    }

    public render(): JSX.Element {
        const failedRuleIds: string[] = Object.keys(this.props.failedRules);
        const ruleName: string = failedRuleIds[this.state.currentRuleIndex];
        const rule: DecoratedAxeNodeResult = this.props.failedRules[ruleName];

        if (this.props.featureFlagStoreData[FeatureFlags.shadowDialog]) {
            return this.withshadowDomTurnedOn(rule);
        } else {
            return this.withshadowDomTurnedOff(rule);
        }
    }

    private getOnClickWhenNotInShadowDom(func: (ev: any) => void): (ev: any) => void {
        if (this.props.featureFlagStoreData[FeatureFlags.shadowDialog]) {
            return null;
        } else {
            return func;
        }
    }

    private renderButtonContainer(): JSX.Element {
        return (
            <div className="insights-dialog-target-button-container">
                {this.renderInspectButton()}
                {this.renderIssueButtons()}
                {this.renderInspectMessage()}
            </div>
        );
    }

    private renderInspectButton(): JSX.Element {
        return (
            <DefaultButton
                className="insights-dialog-button-inspect"
                disabled={this.props.dialogHandler.isInspectButtonDisabled(this)}
                iconProps={{ iconName: 'FileHTML' }}
                text="Inspect HTML"
                onClick={this.getOnClickWhenNotInShadowDom(this.onClickInspectButton)}
            />
        );
    }

    private renderCreateBugButton(issueData: CreateIssueDetailsTextData): JSX.Element {
        return (
            <FileIssueDetailsButton
                deps={this.props.deps}
                issueDetailsData={issueData}
                issueTrackerPath={this.state.issueTrackerPath}
                restoreFocus={false}
            />
        );
    }

    private renderIssueButtons(): JSX.Element {
        const failedRuleIds: string[] = Object.keys(this.props.failedRules);
        const ruleName: string = failedRuleIds[this.state.currentRuleIndex];
        const ruleResult: DecoratedAxeNodeResult = this.props.failedRules[ruleName];
        const issueData: CreateIssueDetailsTextData = {
            pageTitle: document.title,
            pageUrl: document.URL,
            ruleResult,
        };

        return (
            <>
                <CopyIssueDetailsButton
                    deps={this.props.deps}
                    issueDetailsData={issueData}
                    onClick={this.props.deps.targetPageActionMessageCreator.copyIssueDetailsClicked}
                />
                <FlaggedComponent
                    featureFlagStoreData={this.props.featureFlagStoreData}
                    featureFlag={FeatureFlags.showBugFiling}
                    enableJSXElement={this.renderCreateBugButton(issueData)}
                />
            </>
        );
    }

    private renderInspectMessage(): JSX.Element {
        if (this.props.dialogHandler.isInspectButtonDisabled(this)) {
            return (
                <div className="insights-dialog-inspect-disabled">
                    {`To enable the Inspect HTML button, open the Chrome dev tools (${this.props.devToolsShortcut}).`}
                </div>
            );
        }
    }

    private renderNextAndBackButton(): JSX.Element {
        return (
            <div className="ms-Grid">
                <div className="ms-Grid-row">
                    <div className="ms-Grid-col ms-sm3 ms-md3 ms-lg3 insights-dialog-button-left">
                        <PrimaryButton
                            data-automation-id="back"
                            disabled={this.props.dialogHandler.isBackButtonDisabled(this)}
                            text="< Back"
                            onClick={this.getOnClickWhenNotInShadowDom(this.onClickBackButton)}
                        />
                    </div>
                    <div className="ms-Grid-col ms-sm6 ms-md6 ms-lg6 insights-dialog-footer">
                        <div>{this.props.dialogHandler.getFailureInfo(this)}</div>
                    </div>
                    <div className="ms-Grid-col ms-sm3 ms-md3 ms-lg3 insights-dialog-button-right">
                        <PrimaryButton
                            data-automation-id="next"
                            disabled={this.props.dialogHandler.isNextButtonDisabled(this)}
                            text="Next >"
                            onClick={this.getOnClickWhenNotInShadowDom(this.onClickNextButton)}
                        />
                    </div>
                </div>
            </div>
        );
    }

    private renderRuleContainer(rule: DecoratedAxeNodeResult): JSX.Element {
        const fixUrl = (url: string) => {
            if (url.indexOf('://') >= 0) {
                return url;
            } else {
                const { clientBrowserAdapter } = this.props.deps;
                return clientBrowserAdapter.getUrl(url);
            }
        };

        return (
            <div className="insights-dialog-rule-container">
                <Icon iconName="StatusErrorFull" />
                <div className="ms-fontSize-mPlus insights-dialog-rule-link">
                    Rule name: <NewTabLink href={fixUrl(rule.helpUrl)}>{rule.ruleId}</NewTabLink>
                </div>
            </div>
        );
    }

    private renderFixInstructions(ruleResult: DecoratedAxeNodeResult): JSX.Element {
        return (
            <div className="insights-dialog-fix-instruction-container">
                <FixInstructionPanel checkType={CheckType.All} checks={ruleResult.all.concat(ruleResult.none)} />

                <FixInstructionPanel checkType={CheckType.Any} checks={ruleResult.any} />
            </div>
        );
    }

    private renderTargetContainer(): JSX.Element {
        return (
            <div className="insights-dialog-target-container">
                <div className="ms-fontWeight-semibold">Path:</div>
                <div className="insights-dialog-instance-selector">{this.props.elementSelector}</div>
                {this.renderButtonContainer()}
            </div>
        );
    }

    private renderDialogContent(rule: DecoratedAxeNodeResult): JSX.Element {
        return (
            <div>
                {this.renderRuleContainer(rule)}
                {this.renderTargetContainer()}
                {this.renderFixInstructions(rule)}
                {this.renderNextAndBackButton()}
            </div>
        );
    }

    private withshadowDomTurnedOn(rule: DecoratedAxeNodeResult): JSX.Element {
        return (
            <div style={{ visibility: this.state.showDialog ? 'visible' : 'hidden' }} className="insights-dialog-main-override-shadow">
                <div className="insights-dialog-container">
                    <div className="insights-dialog-header">
                        <p className="ms-Dialog-title">{rule.help}</p>
                        <div className="ms-Dialog-topButton">
                            <button
                                type="button"
                                className="ms-Dialog-button ms-Dialog-button--close ms-Button ms-Button--icon insights-dialog-close"
                                aria-label="Close"
                                data-is-focusable="true"
                            >
                                <div className="ms-button-flex-container">
                                    <Icon iconName="Cancel" />
                                </div>
                            </button>
                        </div>
                    </div>

                    {this.renderDialogContent(rule)}
                </div>
            </div>
        );
    }

    private withshadowDomTurnedOff(rule: DecoratedAxeNodeResult): JSX.Element {
        return (
            <Dialog
                hidden={!this.state.showDialog}
                dialogContentProps={{
                    type: DialogType.normal,
                }}
                modalProps={{
                    isBlocking: false,
                    containerClassName: 'insights-dialog-main-container',
                    layerProps: {
                        onLayerDidMount: this.onLayoutDidMount,
                        className: 'insights-dialog-main-override',
                        hostId: 'insights-dialog-layer-host',
                    },
                }}
                onDismiss={this.onHideDialog}
                title={rule.help}
            >
                {this.renderDialogContent(rule)}
            </Dialog>
        );
    }
}
