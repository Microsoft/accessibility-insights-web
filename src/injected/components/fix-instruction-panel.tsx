// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { autobind } from '@uifabric/utilities';
import { Label } from 'office-ui-fabric-react/lib/Label';
import * as React from 'react';
import { CheckType } from './details-dialog';

export interface FixInstructionPanelProps {
    checkType: CheckType;
    checks: FormattedCheckResult[];
}

export class FixInstructionPanel extends React.Component<FixInstructionPanelProps, any> {
    public render(): JSX.Element {
        if (this.props.checks.length === 0) {
            return null;
        }
        const title: string = this.getPanelTitle(this.props.checkType, this.props.checks.length);

        return (
            <div>
                <div className="insights-fix-instruction-title">{title}</div>
                <ul className="insights-fix-instruction-list">{this.renderInstructions(this.props.checkType)}</ul>
            </div>
        );
    }

    private getPanelTitle(checkType: CheckType, checkCount: number): string {
        if (checkCount === 1) {
            return 'Fix the following:';
        }
        if (checkType === CheckType.Any) {
            return 'Fix ONE of the following:';
        } else {
            return 'Fix ALL of the following:';
        }
    }

    private renderInstructions(checkType: CheckType): JSX.Element[] {
        const instructionList = this.props.checks.map((check, checkIndex) => {
            return <li key={`instruction-${CheckType[checkType]}-${checkIndex + 1}`}>{check.message}</li>;
        });
        return instructionList;
    }
}
