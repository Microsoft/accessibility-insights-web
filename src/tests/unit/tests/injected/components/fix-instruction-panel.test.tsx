// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import * as React from 'react';

import { CheckType } from '../../../../../injected/components/details-dialog';
import { FixInstructionPanel, FixInstructionPanelProps } from '../../../../../injected/components/fix-instruction-panel';
import { ShallowRenderer } from '../../../common/shallow-renderer';

describe('FixInstructionPanelTests', () => {
    test('render all checks', () => {
        const allchecks: FormattedCheckResult[] = [
            {
                message: 'message',
                id: 'id1',
                data: 'data',
            },
            {
                message: 'message',
                id: 'id2',
                data: 'data',
            },
        ];

        const props: FixInstructionPanelProps = {
            checkType: CheckType.All,
            checks: allchecks,
        };

        const testObject = new FixInstructionPanel(props);
        const expected = (
            <div>
                <div className="insights-fix-instruction-title">Fix ALL of the following:</div>
                <ul className="insights-fix-instruction-list">
                    <li key={'instruction-All-1'}>message</li>
                    <li key={'instruction-All-2'}>message</li>
                </ul>
            </div>
        );

        expect(testObject.render()).toEqual(expected);
    });

    test('render instruction only one check', () => {
        const allchecks: FormattedCheckResult[] = [
            {
                message: 'message',
                id: 'id1',
                data: 'data',
            },
        ];

        const props: FixInstructionPanelProps = {
            checkType: CheckType.All,
            checks: allchecks,
        };

        const component = ShallowRenderer.render(FixInstructionPanel, props);
        const panelTitle = component.props.children[0];

        expect(panelTitle.props.children).toBe('Fix the following:');
    });

    test('render none checks', () => {
        const noneChecks: FormattedCheckResult[] = [
            {
                message: 'message',
                id: 'id1',
                data: 'data',
            },
            {
                message: 'message',
                id: 'id2',
                data: 'data',
            },
        ];

        const props: FixInstructionPanelProps = {
            checkType: CheckType.None,
            checks: noneChecks,
        };

        const testObject = new FixInstructionPanel(props);
        const expected = (
            <div>
                <div className="insights-fix-instruction-title">Fix ALL of the following:</div>
                <ul className="insights-fix-instruction-list">
                    <li key={'instruction-None-1'}>message</li>
                    <li key={'instruction-None-2'}>message</li>
                </ul>
            </div>
        );

        expect(testObject.render()).toEqual(expected);
    });

    test('render nothing', () => {
        const checks: FormattedCheckResult[] = [];

        const props: FixInstructionPanelProps = {
            checkType: CheckType.None,
            checks: checks,
        };

        const component = ShallowRenderer.render(FixInstructionPanel, props);
        expect(component).toBeNull();
    });
});
