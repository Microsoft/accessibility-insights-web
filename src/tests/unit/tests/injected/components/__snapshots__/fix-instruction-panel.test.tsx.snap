// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FixInstructionPanelTests render all checks 1`] = `
<div>
  <div
    className="insights-fix-instruction-title"
  >
    Fix ALL of the following:
  </div>
  <ul
    className="insights-fix-instruction-list"
  >
    <li>
      message
    </li>
    <li>
      message
    </li>
  </ul>
</div>
`;

exports[`FixInstructionPanelTests render instruction only one check 1`] = `
<div>
  <div
    className="insights-fix-instruction-title"
  >
    Fix the following:
  </div>
  <ul
    className="insights-fix-instruction-list"
  >
    <li>
      message
    </li>
  </ul>
</div>
`;

exports[`FixInstructionPanelTests render none checks 1`] = `
<div>
  <div
    className="insights-fix-instruction-title"
  >
    Fix ALL of the following:
  </div>
  <ul
    className="insights-fix-instruction-list"
  >
    <li>
      message
    </li>
    <li>
      message
    </li>
  </ul>
</div>
`;

exports[`FixInstructionPanelTests render nothing 1`] = `null`;
